### PR TITLE
Package p5scm.0.2.0

### PR DESCRIPTION
**Pull request by drjdn**
### `p5scm.0.2.0`
Scheme via camlp5
This a simple scheme implementation using pa_schemer from camlp5

---
* Homepage: https://github.com/drjdn/p5scm
* Source repo: git+https://github.com/drjdn/p5scm.git
* Bug tracker: https://github.com/drjdn/p5scm/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by mseri** [comment]
There are two lower-bounds failure, with ocaml 4.12 and 4.13. Do you know which package needs a lower bound to fix this:
```

#=== ERROR while compiling p5scm.0.2.0 ========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.12.1 | pinned(https://github.com/drjdn/p5scm/archive/0.2.0.tar.gz)
# path                 ~/.opam/4.12/.opam-switch/build/p5scm.0.2.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p p5scm -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/p5scm-21975-6e9398.env
# output-file          ~/.opam/log/p5scm-21975-6e9398.out
### output ###
# File "/home/opam/.opam/4.12/lib/camomile/camomile.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/camomile/default_config/camomile.default_config.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/camomile/dyn/camomile.dyn.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/camomile/lib_default/camomile.lib_default.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/camomile/library/camomile.library.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/lwt_log/lwt_log.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/lwt_log/core/lwt_log.core.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/re.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/emacs/re.emacs.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/glob/re.glob.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/pcre/re.pcre.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/perl/re.perl.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/posix/re.posix.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/re/str/re.str.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/ocaml-compiler-libs/bytecomp/ocaml-compiler-libs.bytecomp.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/ocaml-compiler-libs/common/ocaml-compiler-libs.common.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/ocaml-compiler-libs/shadow/ocaml-compiler-libs.shadow.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/ocaml-compiler-libs/toplevel/ocaml-compiler-libs.toplevel.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.12/lib/ppx_derivers/ppx_derivers.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
#         make src/lib/exparser.{ml,mli},src/lib/pa_scheme.ml,src/lib/pconfig.ml,src/lib/pr_dump.ml,src/lib/pr_o.ml (exit 2)
# (cd _build/default/src/lib/camlp5 && /usr/bin/make)
# cppo pa_schemer.cppo.ml -o pa_schemer.ml
# camlp5 pa_r.cmo pa_rp.cmo pa_extend.cmo q_MLast.cmo pr_o.cmo pa_schemer.ml -o ../pa_scheme.ml
# cppo pr_o.cppo.ml -o pr_o.ml
# camlp5 pa_r.cmo pa_rp.cmo pa_macro.cmo q_MLast.cmo pa_extfun.cmo pa_extprint.cmo pa_pprintf.cmo pr_o.cmo pr_o.ml -o ../pr_o.ml
# File "pr_o.ml", line 638, characters 46-47:
# While expanding quotation "constructor":
# Parse error: end of input expected after [constructor_declaration] (in
#   [constructor_eoi])
# make: *** [Makefile:14: pr_o] Error 2
```

This is what the CI is doing to trigger the issue:
```
The following actions will be performed:
  - remove    camlp-streams               5.0                  [conflicts with dune]
  - install   jbuilder                    1.0+beta20.1         [required by ocaml-compiler-libs, camomile, ppx_derivers]
  - remove    csexp                       1.5.1                [conflicts with dune]
  - remove    opam-depext                 1.2.1
  - downgrade ocamlfind                   1.9.3 to 1.9.1
  - downgrade conf-perl                   2 to 1
  - downgrade result                      1.5 to 1.0
  - downgrade ppx_derivers                1.2.1 to 1.0
  - downgrade ocaml-compiler-libs         v0.12.4 to v0.11.0
  - downgrade camomile                    1.0.2 to 1.0.1
  - downgrade dune                        2.9.3 to 2.8.5
  - recompile base-bytes                  base                 [uses ocamlfind]
  - downgrade topkg                       1.0.5 to 1.0.3       [uses ocamlfind]
  - downgrade react                       1.2.1 to 1.0.0       [uses ocamlfind]
  - downgrade num                         1.4 to 1.0           [uses ocamlfind]
  - downgrade conf-perl-string-shellquote 2 to 1               [uses conf-perl]
  - downgrade conf-perl-ipc-system-simple 2 to 1               [uses conf-perl]
  - install   ocaml-migrate-parsetree     2.2.0                [required by ppxlib]
  - install   dune-private-libs           2.3.0                [required by dune-configurator]
  - recompile trie                        1.0.0                [uses dune]
  - recompile sexplib0                    v0.14.0              [uses dune]
  - recompile mmap                        1.1.0                [uses dune]
  - recompile charInfo_width              1.1.0                [uses dune, camomile, result]
  - downgrade stdlib-shims                0.3.0 to 0.1.0       [uses dune]
  - downgrade seq                         base to 0.2.2        [uses dune]
  - downgrade menhirSdk                   20211230 to 20201214 [uses dune]
  - downgrade menhirLib                   20211230 to 20201214 [uses dune]
  - downgrade cppo                        1.6.8 to 1.6.6       [uses dune]
  - downgrade octavius                    1.2.2 to 0.1.0       [uses ocamlfind]
  - downgrade camlp5                      8.00.03 to 8.00.02   [uses conf-perl]
  - downgrade dune-configurator           2.9.3 to 2.3.0       [uses dune]
  - recompile mew                         0.1.0                [uses dune, result]
  - recompile zed                         3.1.0                [uses camomile, dune]
  - downgrade ppxlib                      0.24.0 to 0.22.2     [uses ocaml-compiler-libs, dune, ppx_derivers]
  - downgrade re                          1.10.3 to 1.8.0      [uses jbuilder]
  - downgrade menhir                      20211230 to 20201214 [uses dune]
  - downgrade ocplib-endian               1.2 to 1.0           [uses ocamlfind]
  - downgrade base                        v0.14.3 to v0.14.1   [uses dune]
  - recompile mew_vi                      0.5.0                [uses dune]
  - downgrade lwt                         5.5.0 to 5.4.0       [uses dune, result]
  - recompile ppx_sexp_conv               v0.14.3              [uses dune]
  - recompile ppx_js_style                v0.14.1              [uses dune]
  - recompile ppx_enumerate               v0.14.0              [uses dune]
  - recompile ppx_compare                 v0.14.0              [uses dune]
  - recompile ppx_cold                    v0.14.0              [uses dune]
  - downgrade parsexp                     v0.14.2 to v0.14.1   [uses dune]
  - downgrade lwt_react                   1.1.5 to 1.0.1       [uses lwt]
  - downgrade lwt_log                     1.1.1 to 1.1.0       [uses jbuilder]
  - recompile ppx_hash                    v0.14.0              [uses dune]
  - recompile sexplib                     v0.14.0              [uses dune]
  - recompile lambda-term                 3.1.0                [uses camomile, dune]
  - recompile ppx_base                    v0.14.0              [uses dune]
  - downgrade utop                        2.9.0 to 2.8.0       [uses dune, camomile, ocamlfind]
  - recompile sexp_pretty                 v0.14.0              [uses dune]
===== 3 to install | 18 to recompile | 30 to downgrade | 3 to remove =====
```

Full log, e.g., here: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/d9acf4f7db3ab4462c34273d56727557d96d8f32/variant/opam-2.1,compilers,4.13,p5scm.0.2.0,lower-bounds

File: packages/p5scm/p5scm.0.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+authors: ["Jason D. Nielsen"]
+license: "MIT"
+homepage: "https://github.com/drjdn/p5scm"
+bug-reports: "https://github.com/drjdn/p5scm/issues"
+depends: [
+  "dune" {>= "2.8"}
+  "ocaml" {>= "4.12.0" & < "4.15.0"}
+  "menhir" {>= "20201214"}
+  "cppo" {>= "1.6.6"}
+  "camlp5" {>= "8.00~alpha05"}
```
Comment (mseri) [inline packages/p5scm/p5scm.0.2.0/opam:16]:
```suggestion
  "camlp5" {>= "8.00.03"}
```

**Comment by mseri** [comment]
Perfect. It is enough to fix it here, without the need to rerelease. Please port the change upstream

**Comment by drjdn** [comment]
Thanks.  Already fixed in the github repo.

**Comment by mseri** [comment]
Oh ok, can you republish then. The checksum needs update now!
```
[ERROR] The sources of the following couldn't be obtained, aborting:
          - p5scm.0.2.0:
            https://github.com/drjdn/p5scm/archive/0.2.0.tar.gz (Bad checksum, expected md5=f2f68c242a46558c03f955f957d62d41)
            ```

**Comment by mseri** [comment]
Thanks!

**Pull request merged**